src/main/java/org/troyargonauts/subsystems/Elevator.java

```diff
@@ -0,0 +1,125 @@
+//
+//        lowerLimitSwitch = leftMotor.getForwardLimitSwitch(SparkMaxLimitSwitch.Type.kNormallyClosed);
+//        lowerLimitSwitch = rightMotor.getForwardLimitSwitch(SparkMaxLimitSwitch.Type.kNormallyClosed);
+        pid = new PIDController(Constants.Elevator.kP, Constants.Elevator.kI ,Constants.Elevator.kD, Constants.Elevator.PERIOD);
+    }
+
+    /**
+     * Periodic will constantly check the encoder and limit switch values.
+     * These values will later be used to make sure elevator extends or retracts to optimal position.
+     */
+    @Override
+    public void periodic() {
+//        upperLimitSwitchValue = getUpperLimitSwitchState();
+//        lowerLimitSwitchValue = getLowerLimitSwitchState();
+        SmartDashboard.putNumber("Right Encoder", rightMotor.getEncoder().getPosition());
+        SmartDashboard.putNumber("Left Encoder", leftMotor.getEncoder().getPosition());
+
+        SmartDashboard.putNumber("Left Draw", leftMotor.getOutputCurrent());
+        SmartDashboard.putNumber("Right Draw", rightMotor.getOutputCurrent());
+    }
+
+    // /**
+    //  * Will check whether the upper limit switch is enabled or not.
+    //  * If it is, this means the elevator has extended to desired point and will stop extending
+    //  *
+    //  * @return  returns whether the limit switch is enabled or not.
+    //  */
+    // public boolean getUpperLimitSwitchState() {
+    //     return upperLimitSwitch.isLimitSwitchEnabled();
+    // }
+    // /**
+    //  * Will check whether the lower limit switch is enabled or not.
+    //  * If it is, this means the elevator has retracted to desired point and will stop retracting
+    //  *
+    //  * @return  returns whether the limit switch is enabled or not.
+    //  */
+    // public boolean getLowerLimitSwitchState() {
+    //     return lowerLimitSwitch.isLimitSwitchEnabled();
+    // }
+
+    /**
+     * Elevator power will be set to a given speed from -1 to 1
+     * If it is positive, elevator will extend until upper limit switch turns on
+     * If it is negative, elevator retracts until lower limit switch turns on
+     *
+     * @param speed desired elevator extension or retraction speed
+     */
+    public void setPower(double speed, double nerf) {
+        leftMotor.set(speed * nerf);
+        rightMotor.set(speed * nerf);
+    }
+
+    /** 
+     * Returns encoder position based on encoder values
+     * @return encoder position based on encoder values
+     */
+    public double getPosition() {
+        return (rightMotor.getEncoder().getPosition() + leftMotor.getEncoder().getPosition()) / (2 * Constants.Elevator.kEncoderGearboxScale);
+    }
+
+    /**
+     * Using a PID command, the elevator will shift to a given setpoint using the
+     * predetermined PID Controller.
+     * 
+     * @param setpoint will be the desired extension point
+     */
+    public void elevatorPID(double setpoint) {
```

Comment:
Set PIDController's setpoint within method

Reply:
Refer to line 175 in [Drivetrain.java](https://github.com/Troy-Argonauts/2023RobotCode/blob/feature/drivetrain-pid/src/main/java/org/troyargonauts/subsystems/DriveTrain.java) on feature/drivetrain-pid branch for an example.